src/unravel/core.cljs

```diff
@@ -17,30 +17,85 @@
 
 (defn init [])
 
-(spec/def ::cmdline-args
-  (spec/cat
-   :options (spec/* (spec/alt :version #{"--version"}
-                              :debug #{"--debug"}
-                              :cp (spec/& (spec/cat :_ #{"--classpath" "-c"} :path string?) (spec/conformer #(:path %)))
-                              :blobs (spec/& (spec/cat :_ #{"--blob"} :path string?) (spec/conformer #(:path %)))
-                              :flags (spec/& (spec/cat :_ #{"--flag"} :path string?) (spec/conformer #(:path %)))))
-   :host (spec/? string?) :port (spec/and string? #(re-matches #"\d+" %))))
+(defn parse-arg [m [arg nxt :as args]]
+  (let [switch (fn [test-fn kw]
+                 (when (test-fn arg)
+                   [(assoc m kw true) (rest args)]))
+        mult (fn mult
+               ([test-fn kw] (mult test-fn kw [] identity))
+               ([test-fn kw empty-coll] (mult test-fn kw empty-coll identity))
+               ([test-fn kw empty-coll val-fn]
+                (when (test-fn arg)
+                  (assert (some? nxt) "Needs parameter")
+                  [(update m kw (fn [elements] (conj (or elements empty-coll) (val-fn nxt)))) (rest (rest args))])))]
+    (or
+     (switch #{"--version"} :version?)
+     (switch #{"--debug"} :debug?)
+     (mult #{"--classpath" "-c"} :cp)
+     (mult #{"--blob"} :blobs)
+     (mult #{"--flag"} :flags #{} keyword))))
+
+(defn parse-args [args]
+  (loop [m {}
+         [arg :as args] args]
+    (if-let [[m* args*] (parse-arg m args)]
+      (recur m* args*)
+      (cond
+        (nil? arg)
+        m
+
+        (.startsWith arg "-")
+        (throw (js/Error. (str "Unknown argument: " arg)))
+
+        :else
+        (recur (update m :positional (fn [xs] (conj (or xs []) arg)))
+               (rest args))))))
 
 (def help-text
-  "Syntax: unravel [--debug] [-c|--classpath <paths>] [--blob blob1 [--blob blob2 ...]] [--flag flag1 [--flag --flag2 ...]] [<host>] <port>\n        unravel --version")
+  "Syntax: unravel [--debug] [-c|--classpath <paths>] [--blob blob1 [--blob blob2 ...]] [--flag flag1 [--flag --flag2 ...]] [<host> <port>]\n        unravel --version")
+
+(defn jack-in [cb]
+  (let [child (.spawn (js/require "child_process")
+                      "bash"
+                      #js [(un/locate "scripts/jack-in")])
+        waiting? (atom true)
+        terminate-fn (fn terminate-fn []
+                       (.kill child))
+        sb (StringBuffer.)]
+    (-> child
+        (.on "exit" (fn []
+                      (-> child .-stdout .destroy)
```

Comment:
@cgrand wondering if this is the right thing to do. But without this, node doesn't exit. Not sure what's going on there